test/RequestResponseTest.cpp

```diff
@@ -135,6 +135,28 @@ TEST(RequestResponseTest, FailureInResponse) {
   to->assertOnErrorMessage("whew!");
 }
 
+TEST(RequestResponseTest, RequestOnDisconnectedClient) {
+  folly::ScopedEventBaseThread worker;
+  auto client = makeDisconnectedClient(worker.getEventBase());
+
+  auto requester = client->getRequester();
+  bool did_call_on_error = false;
+  folly::Baton<> wait_for_on_error;
+  requester->requestResponse(Payload("foo", "bar"))
+      ->subscribe(
+          [](auto) {
+            // should not call onSuccess
+            FAIL();
+          },
+          [&](folly::exception_wrapper) {
+            did_call_on_error = true;
+            wait_for_on_error.post();
+          });
```

Comment:
Did you want to do an onComplete lambda that also does `FAIL()` (mimicking `RequestStreamTest.RequestOnDisconnectedClient`)

Reply:
It's requestResponse, so the subscriber only has onSuccess and onError

Reply:
Oh right, I always forget that...